app/controllers/articles_controller.rb

```diff
@@ -1,54 +1,55 @@
 class ArticlesController < ApplicationController
 
-  http_basic_authenticate_with name: "sebas", password: "secret", except: [:index, :show]
+  include PrivateGuard
+  include SecureTransaction
+
+  before_action :require_user
 
   def index
     @articles = Article.all
   end
 
   def show
-    @article = Article.find(params[:id])
+    @article = find(Article, params[:id]) # Article.find(params[:id])
   end
 
   def new
     @article = Article.new
   end
 
   def create
-    p article_params
-    @article = Article.new(article_params)
-    if @article.save
+    @article = current_user.articles.create(article_params)
+    if @article
       redirect_to @article
     else
       render :new
     end
   end
 
   def edit
-    @article = Article.find(params[:id])
+    @article = find(current_user.articles, params[:id]) # current_user.articles.find(params[:id])
   end
 
   def update
-    @article = Article.find(params[:id])
-
-    if @article.update(article_params)
-      redirect_to @article
-    else
-      render :edit
+    if (@article = find(current_user.articles, params[:id])).is_a?(Article) # current_user.articles.find(params[:id])
```

Comment:
@JsMontesR too much logic to be placed in a Controller, moving logic into a Class is a common pattern ... just as a tip.